src/main/java/org/sagebionetworks/bridge/models/files/FileReference.java

```diff
@@ -0,0 +1,78 @@
+package org.sagebionetworks.bridge.models.files;
+
+import static org.sagebionetworks.bridge.config.Environment.PROD;
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.config.Environment;
+import org.sagebionetworks.bridge.json.DateTimeSerializer;
+
+public final class FileReference {
+    
+    private final Environment env;
+    private final String baseUrl;
+    private final String guid;
+    private final DateTime createdOn;
+    
+    @JsonCreator
+    public FileReference(@JsonProperty("guid") String guid, @JsonProperty("createdOn") DateTime createdOn) {
+        this.guid = guid;
+        this.createdOn = (createdOn == null) ? null : createdOn.withZone(DateTimeZone.UTC);
+        this.env = BridgeConfigFactory.getConfig().getEnvironment();
+        this.baseUrl = BridgeConfigFactory.getConfig().getHostnameWithPostfix("docs");
+    }
```

Comment:
If we don't want to use PowerMock, how about using a FileReferenceFactory?

Reply:
I don't see how a factory would be better than a constructor. Switched to PowerMock. It doesn't run in Eclipse but it does run from command line.